packages/react-reconciler/src/ReactFiberLazyComponent.js

```diff
@@ -47,6 +47,7 @@ export function readLazyComponentType<T>(lazyComponent: LazyComponent<T>): T {
       lazyComponent._status = Pending;
       const ctor = lazyComponent._ctor;
       const thenable = ctor();
+      lazyComponent._result = thenable;
```

Comment:
This is still needed because we read thenable from `lazyComponent._result` while it's pending. If we keep the assignment, it will overwrite sync thenables. But if we remove it at all, we will throw `null` instead of thenable. This fixes both cases.

Reply:
You still need the assignment, but I don't think you need to move it up here. If you keep it right after the early return branch you added on line 78-80, then there's always a single assignment.

```js
if (lazyComponent._status === Resolved) {
  return lazyComponent._result;
}
lazyComponent._result = thenable;
throw thenable;
```